commit-message.md

```diff
@@ -0,0 +1,36 @@
+# Guidelines to create great commit messages
+
+
+We use [GitCop](https://gitcop.com) to check that commit messages are
+properly written. The rules are the following:
+
+* The first line of a commit message, called the subject line should
+  not be more than 80 characters long.
+
+* The commit message should end with the following trailers:
+
+  ```
+  Licence: MIT
+  Signed-off-by: User Name <email@address>
+  ```
+
+  where "User Name" is the author's real name and email@address one of
```

Comment:
If it were practical *today* then you would have already seen contribution/review signing in linux kernel / any high stake project DCO (as one of the sufficient conditions of being practical).
(b), (c) are among the real obstacles to contribution/review signing, which has nothing to do with the lack of motivation (neither is this the reason for the state of email+pgp).

> > When it becomes practical it will enable pseudonymous contributor/reviewer to
 verify itself.
> For this purpose, I think it's practical now.

I think you are referring to 1. pseudonymous contributor/reviewer for sign-off or commit author field + pgp-signed emails / fake real name, but those are not *signed commits*, 2. one/few author(s) case where merges are unneeded/manageable, then sure this is practical and has been done for a very long time.

For the second (a) case, so given the choice, you are substituting a permanent pki gov id with a pki + one-off 2nd factor auth. Why?
For the second (b) case, this is what US law http://www.copyright.gov/fls/fl101.html says
> In no case should you omit the name of the copyright claimant. You can use a pseudonym for the claimant name. But be aware that if a copyright is held under a fictitious name, business dealings involving the copyrighted property may raise questions about its ownership. Consult an attorney for legal advice on this matter.

So it is up to the lawyers.